products/accesscontextmanager/api.yaml

```
@@ -260,3 +260,154 @@ objects:
+          perimeter content and boundaries.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: 'resources'
+            description: |
+              A list of GCP resources that are inside of the service perimeter.
+              Currently only projects are allowed.
+              Format: projects/{project_number}
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: 'accessLevels'
+            description: |
+              A list of AccessLevel resource names that allow resources within
+              the ServicePerimeter to be accessed from the internet.
+              AccessLevels listed must be in the same policy as this
+              ServicePerimeter. Referencing a nonexistent AccessLevel is a
+              syntax error. If no AccessLevel names are listed, resources within
+              the perimeter can only be accessed via GCP calls with request
+              origins within the perimeter. For Service Perimeter Bridge, must
+              be empty.
+
+              Format: accessPolicies/{policy_id}/accessLevels/{access_level_name}
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: 'unrestrictedServices'
+            description: |
+              GCP services that are not subject to the Service Perimeter
+              restrictions. May contain a list of services or a single wildcard
+              "*". For example, if logging.googleapis.com is unrestricted, users
+              can access logs inside the perimeter as if the perimeter doesn't
+              exist, and it also means VMs inside the perimeter can access logs
+              outside the perimeter.
+
+              The wildcard means that unless explicitly specified by
+              "restrictedServices" list, any service is treated as unrestricted.
+              One of the fields "restrictedServices", "unrestrictedServices"
```

Do we usually try to add validation for things like this or just let the api error out since it's pretty clearly documented?

---

Just let it error out - we have no good means of validating multiple-field rules